sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1044,6 +1044,34 @@ class Analyzer(
           failOnOuterReference(p)
           p
       }
+
+      // SPARK-17348
+      // Looking for a potential incorrect result case.
+      // When a correlated predicate is a non-equality predicate
+      // it must be placed at the immediate child operator.
+      // Otherwise, the pull up of the correlated predicate
+      // will generate a plan with a different semantics
+      // which could return incorrect result.
+      var continue : Boolean = true
```

Comment:
That would regress a scenario currently support. The following queries demonstrate the good case that is working (giving correct results) today.

Seq((1,1)).toDF("c1","c2").createOrReplaceTempView("t1")
Seq((1,1),(2,0)).toDF("c1","c2").createOrReplaceTempView("t2")
sql("select c1 from t1 where c1 in (select t2.c1 from t2 where t1.c2 >= t2.c2)").show
sql("select t1.c1 from t1, t1 as t3 where t1.c1 = t3.c1 and (t1.c1 in (select t2.c1 from t2 where t1.c2 >= t2.c2 or t3.c2 < t2.c2) or t1.c2 >= 0)").show

Those two positive test cases have been added into SubquerySuite as part of this PR.

We would want to extend the support blocked by this PR in the future once we implement the correlated subquery processing.